modules/apps/27-interchain-accounts/controller/keeper/handshake_test.go

```diff
@@ -294,11 +294,14 @@ func (suite *KeeperTestSuite) TestOnChanOpenInit() {
 					Version:        string(versionBytes),
 				}
 
-				chanCap, err = suite.chainA.App.GetScopedIBCKeeper().NewCapability(suite.chainA.GetContext(), host.ChannelCapabilityPath(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID))
-				suite.Require().NoError(err)
+				channelID := channeltypes.FormatChannelIdentifier(suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetNextChannelSequence(suite.chainA.GetContext()))
+				path.EndpointA.ChannelID = channelID
 
 				tc.malleate() // malleate mutates test data
 
+				chanCap, err = suite.chainA.App.GetScopedIBCKeeper().NewCapability(suite.chainA.GetContext(), host.ChannelCapabilityPath(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID))
+				suite.Require().NoError(err)
+
```

Comment:
I find it weird that I had to do this to fix failing tests here given the diffs, I can try to provide more info later, need more reasoning on it

Reply:
Let me try to explain this a little further.

The channel capability is generated in 04-channel (core IBC) and passed to application callback handlers (e.g. like this one under test - ica `OnChanOpenInit`).
This test calls `keeper.OnChanOpenInit` directly, rather than delivering a msg e2e, testing core integration points with the app - thus, the capability is generated directly in the test code here and passed to app callback.

From what I can see, this test as it existed, was creating a new capability with an empty channel ID. This is because the `path.EndpointA.ChannelID` field is not actually fulfilled here until after the `INIT` step of the handshake. 

In the diffs, I get the next channel sequence nonce (monotonically increasing) and format a new channel ID with it. Then use that to create the new capability before invoking the app callbacks directly on the keeper.
It's moved under the call to `tc.malleate()` as some test cases create a channel and set it to closed, in order to test regaining control of an interchain account on a closed channel, i.e. without moving it we fix some tests, and cause failures in others surfaced from capability already being claimed for a particular channelID.

Hope that makes sense!

Reply:
The changes which caused the test failures to be raised in this PR are from `GetActiveChannelID` which previously allowed storing an empty string and have it being treated as a "found" or "truthy" value.

e.g. in testcase `failure: different ordering from previous channel`, there is a call to `SetActiveChannelID` in the `malleate()` func. It's actually setting an empty channel id as the active channel id on main, while `store.Has()` treats it as true.